frame/contracts/src/lib.rs

```diff
@@ -975,6 +1000,19 @@ trait Invokable<T: Config> {
 		environmental!(executing_contract: bool);
 
 		let gas_limit = common.gas_limit;
+
+		// Check whether the origin is allowed here. The responsibility of making this decision is
+		// delegated to `ensure_origin`, this could vary for different implementations of this
```

Comment:
```suggestion
		// Check whether the origin is allowed here. The logic of the access rules
		// is in the `ensure_origin`, this could vary for different implementations of this
```